### PR TITLE
Test for snapshot corruption when restoring from backup.

### LINKED ISSUE #19984: Snapshot detected as corrupted because of incorrect checksum comparison
**Describe the bug**
On restart of broker
We see 
```
Expected snapshot /usr/local/zeebe/data/raft-partition/partitions/3/snapshots/90926-1-821751-758053 to have checksum 1612185047, but the actual checksum is 3972871514; the snapshot is most likely corrupted. The startup will fail if there is no other valid snapshot and the log has been compacted.
```
then the Partition cannot startup
```
java.lang.IllegalStateException: In partition 2 expected to find a snapshot at index >= log's first index 211099, but found snapshot 0. A previous snapshot is most likely corrupted.
	at io.atomix.raft.utils.StateUtil.verifySnapshotLogConsistent(StateUtil.java:33) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.impl.RaftContext.<init>(RaftContext.java:220) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:249) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:223) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.partition.impl.RaftPartitionServer.buildServer(RaftPartitionServer.java:179) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.partition.impl.RaftPartitionServer.<init>(RaftPartitionServer.java:98) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.partition.RaftPartition.createServer(RaftPartition.java:133) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.partition.RaftPartition.initServer(RaftPartition.java:116) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.atomix.raft.partition.RaftPartition.bootstrap(RaftPartition.java:96) ~[zeebe-atomix-cluster-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.broker.partitioning.startup.steps.RaftBootstrapStep.startup(RaftBootstrapStep.java:34) ~[zeebe-broker-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.broker.partitioning.startup.steps.RaftBootstrapStep.startup(RaftBootstrapStep.java:14) ~[zeebe-broker-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.startup.StartupProcess.proceedWithStartupSynchronized(StartupProcess.java:162) ~[zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3(StartupProcess.java:170) ~[zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:28) ~[zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:96) ~[zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:47) [zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:122) [zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130) [zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108) [zeebe-scheduler-8.5.4.jar:8.5.4]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227) [zeebe-scheduler-8.5.4.jar:8.5.4]
```

On inspecting further, the snapshot is not corrupted. But it is a bug in checksum comparison. 

**Environment:**
- Zeebe Version: 8.5.4

### PR DESCRIPTION
**Pull request by EuroLew**
## Description

When restoring from a backup the snapshot should not be corrupted. To this end a file checksum verification is performed. 

The snapshot store for restoring did not have a checksum provider while all internal backups are taken with a RocksDB checksum provider causing a combined checksum mismatch.  #19988 fixes this however customers that do not have this patch fix will encounter this issue when trying to restore from any snapshots that are created with the help of RocksDB full file checksums.

This PR removes the ability to create a `FileBasedSnapshotStore` without specifying a checksum provider as all stores should aim not to fully read all files when possible for checksums.

`RestoreAcceptance.java` now also restores from snapshots which will then catch the particular case described in paragraph 2. To match the state of the code base to users encountering problems and make the tests fail:
1. `FileBasedSnapshotStore:213` change this line to compare combined values. `if (expectedChecksum.getCombinedValue() != actualChecksum.getCombinedValue()`
2. `PartitionRestoreService:182` Change this snapshot store to just a noop checksum provider `new FileBasedSnapshotStore(
            brokerId,
            partition.id().id(),
            partition.dataDirectory().toPath(),
            snapshotPath -> Map.of());` so it will calculate the checksum by fully reading all files.

## Related issues

closes #19984

### CODE REVIEW

**Comment by lenaschoenburg** [comment]
> This PR removes the ability to create a FileBasedSnapshotStore without specifying a checksum provider as all stores should aim not to fully read all files when possible for checksums.

I don't fully agree with this. Outside of the hotpath (i.e. taking snapshots), it may be valuable to double check file checksums. One such place may be during backup restore.

**Comment by EuroLew** [comment]
In that case we can just provide a no-op or empty checksum provider in my mind this is explicit and better than leaving it empty as it forces the user to think about this step.

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/RestoreAcceptance.java [hunk 1]
```diff
@@ -57,6 +58,9 @@ private void takeBackup(final long backupId) {
         client.newPublishMessageCommand().messageName("name").correlationKey("key").send().join();
       }
 
+      final PartitionsActuator partitions = PartitionsActuator.of(zeebe);
+      partitions.takeSnapshot();
+
```
Comment (deepthidevaki) [inline zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/RestoreAcceptance.java:63]: :x: You should wait until the snapshot is taken before requesting backup. Snapshot is taken asynchronously, so there is a small chance that it is not included in the backup if we don't wait for it. You can use the PartitionActuator to query the snapshot.

```
  Awaitility.await("Snapshot is taken")
        .atMost(Duration.ofSeconds(60))
        .until(
            () ->
                Optional.ofNullable(partitions.query().get(1).snapshotId())
                    .flatMap(FileBasedSnapshotId::ofFileName),
            Optional::isPresent)
```

**Review by deepthidevaki — APPROVED** [review]
Thanks. Please address my comment before merging.

This PR should be also backported so that if we accidentally backport any bugs, the test fails.

**Pull request merged**